### PR TITLE
KEYCLOAK-14140 added more test cases

### PR DESCRIPTION
**Pull request by edewit**
build: https://keycloak-jenkins.com/job/keycloak-account2-matrix/893/

### CODE REVIEW

**Comment by ssilvert** [comment]
@edewit Please squash commits.

File: testsuite/integration-arquillian/tests/other/base-ui/src/test/java/org/keycloak/testsuite/ui/account2/MyResourcesTest.java [hunk 1]
```diff
@@ -30,18 +30,20 @@
...
 import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
 import static org.keycloak.representations.idm.CredentialRepresentation.PASSWORD;
 import static org.keycloak.testsuite.auth.page.AuthRealm.TEST;
 import static org.keycloak.testsuite.util.WaitUtils.pause;
 import static org.keycloak.testsuite.util.WaitUtils.waitForModalFadeIn;
 import static org.keycloak.testsuite.util.WaitUtils.waitForModalFadeOut;
 
-public class MyResourcesTest extends AbstractAccountTest {
+public class MyResourcesTest extends BaseAccountPageTest {
```
Comment (vmuzikar) [inline testsuite/integration-arquillian/tests/other/base-ui/src/test/java/org/keycloak/testsuite/ui/account2/MyResourcesTest.java:46]: Please add some simple notifications checks to all test scenarios, similarly to e.g. [DeviceActivityTest](https://github.com/keycloak/keycloak/blob/de8ba75399209121f4238aef567795c1c4e7c938/testsuite/integration-arquillian/tests/other/base-ui/src/test/java/org/keycloak/testsuite/ui/account2/DeviceActivityTest.java#L158).

Comment (vmuzikar) [inline testsuite/integration-arquillian/tests/other/base-ui/src/test/java/org/keycloak/testsuite/ui/account2/MyResourcesTest.java:46]: Please add a test for the "Application" column – if it's displayed, links are present, etc.

Comment (vmuzikar) [inline testsuite/integration-arquillian/tests/other/base-ui/src/test/java/org/keycloak/testsuite/ui/account2/MyResourcesTest.java:46]: Please add a test for the Pending requests dialog and Edit resource dialog to check if it displays all the users it is supposed to.

**Review by vmuzikar — CHANGES_REQUESTED** [review]
@edewit Thanks for extending the test coverage. Added a few comments for more test scenarios.

File: testsuite/integration-arquillian/tests/other/base-ui/src/test/java/org/keycloak/testsuite/ui/account2/MyResourcesTest.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (edewit) [reply testsuite/integration-arquillian/tests/other/base-ui/src/test/java/org/keycloak/testsuite/ui/account2/MyResourcesTest.java:46]: There is a test to remove scopes, but there is no logic to add a scope to an existing share. Is that a functionality that is missing?

[the remark this answers is not in the document]
Comment (vmuzikar) [reply testsuite/integration-arquillian/tests/other/base-ui/src/test/java/org/keycloak/testsuite/ui/account2/MyResourcesTest.java:46]: Yes, and also to test the Scopes are displayed properly (if it displayes everything that should be displayed).

File: testsuite/integration-arquillian/tests/other/base-ui/src/test/java/org/keycloak/testsuite/ui/account2/MyResourcesTest.java [hunk 2]
```diff
@@ -173,6 +197,23 @@ public void shouldShowMyResourcesAndShare() {
         assertThat(myResourcesPage.getSharedWith(row), endsWith("and 1 other users."));
     }
 
+    @Test
+    public void shouldAllowRequestToShare() {
+        final String resourceName = "Resource 12";
+        myResourcesPage.clickPendingRequest(resourceName);
+        waitForModalFadeIn();
```
[the remark this answers is not in the document]
Comment (edewit) [reply testsuite/integration-arquillian/tests/other/base-ui/src/test/java/org/keycloak/testsuite/ui/account2/MyResourcesTest.java:204]: `testModalDialog` is meant for simple "Are you sure you want to do this" kind of dialog. On these dialog users will have to do more then just press the confirm or cancel buttons.

[the remark this answers is not in the document]
Comment (vmuzikar) [reply testsuite/integration-arquillian/tests/other/base-ui/src/test/java/org/keycloak/testsuite/ui/account2/MyResourcesTest.java:204]: Yes, I agree. But we should use a similar approach to test this more complex dialogs. Like try to close them, check that they are (not) displayed, etc.

**Comment by edewit** [comment]
new build: https://keycloak-jenkins.com/job/keycloak-account2-matrix/894/

**Comment by vmuzikar** [comment]
@edewit Please let me know once this is ready for another review round.

**Comment by edewit** [comment]
@vmuzikar how about now, even rebased it on top of snowpack

File: testsuite/integration-arquillian/tests/other/base-ui/src/test/java/org/keycloak/testsuite/ui/account2/MyResourcesTest.java [hunk 3]
```diff
@@ -131,48 +121,133 @@ public void setup() throws Exception {
...
+        testUser = createUser("jdoe");
+    }
+
+    @Test
+    public void shouldDisplayTheResources() {
+        assertEquals(6, myResourcesPage.getResourcesListCount());
+
+        assertEquals("Resource 0", myResourcesPage.getCellText("name", 0));
+        assertEquals("Resource 1", myResourcesPage.getCellText("name", 1));
+        assertEquals("My Resource Server", myResourcesPage.getCellText("client", 0));
```
Comment (vmuzikar) [inline testsuite/integration-arquillian/tests/other/base-ui/src/test/java/org/keycloak/testsuite/ui/account2/MyResourcesTest.java:161]: Please add a simple assert to verify that link is also present here.
I believe there is some scenario when a link is not present at all, right? We should test that too.

File: testsuite/integration-arquillian/tests/other/base-ui/src/test/java/org/keycloak/testsuite/ui/account2/MyResourcesTest.java [hunk 4]
```diff
@@ -131,48 +121,133 @@ public void setup() throws Exception {
...
+    }
+
     @Test
     public void shouldShowMyResourcesAndUpdatePermissions() {
-        myResourcesPage.navigateTo();
-        loginPage.form().login(createUser("jdoe"));
-        myResourcesPage.assertCurrent();
+        final int row = 1;
+        myResourcesPage.clickExpandButton(row);
```
Comment (vmuzikar) [inline testsuite/integration-arquillian/tests/other/base-ui/src/test/java/org/keycloak/testsuite/ui/account2/MyResourcesTest.java:167]: Could we test that collapsing works as well?

File: testsuite/integration-arquillian/tests/other/base-ui/src/test/java/org/keycloak/testsuite/ui/account2/page/MyResourcesPage.java [hunk 5]
```diff
@@ -34,25 +55,99 @@ public void clickShareButton(int row) {
...
     public void removeAllPermissions() {
         driver.findElement(By.id("remove_pf-random-id-1")).click();
         driver.findElement(By.id("remove_pf-random-id-2")).click();
         driver.findElement(By.id("done")).click();
+        waitForModalFadeOut();
+    }
+
+    public ContentAlert alert() {
+        return alert;
+    }
```
Comment (vmuzikar) [inline testsuite/integration-arquillian/tests/other/base-ui/src/test/java/org/keycloak/testsuite/ui/account2/page/MyResourcesPage.java:148]: Why to override this method? It seems identical (the `alert` field as well) to the parent.

File: testsuite/integration-arquillian/tests/other/base-ui/src/test/java/org/keycloak/testsuite/ui/account2/MyResourcesTest.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (vmuzikar) [reply testsuite/integration-arquillian/tests/other/base-ui/src/test/java/org/keycloak/testsuite/ui/account2/MyResourcesTest.java:47]: I believe this one is still valid. We're missing some test for adding scopes as well as checking that scopes are displayed properly (everything that's ought to be displayed, is displayed).

File: testsuite/integration-arquillian/tests/other/base-ui/src/test/java/org/keycloak/testsuite/ui/account2/MyResourcesTest.java [hunk 6]
```diff
@@ -173,6 +197,23 @@ public void shouldShowMyResourcesAndShare() {
         assertThat(myResourcesPage.getSharedWith(row), endsWith("and 1 other users."));
     }
 
+    @Test
+    public void shouldAllowRequestToShare() {
```
[the remark this answers is not in the document]
Comment (vmuzikar) [reply testsuite/integration-arquillian/tests/other/base-ui/src/test/java/org/keycloak/testsuite/ui/account2/MyResourcesTest.java:221]: I believe this one is still unresolved.

**Review by vmuzikar — CHANGES_REQUESTED** [review]
@edewit Thanks for the update and sorry for delayed response. The coverage is much improved now. Good work, thanks!

Found a few last gaps, other than that looks good.

**Review by Aboullos — DISMISSED** [review]
LGTM

File: testsuite/integration-arquillian/tests/other/base-ui/src/test/java/org/keycloak/testsuite/ui/account2/MyResourcesTest.java (hunk 3 above)
Comment (edewit) [reply testsuite/integration-arquillian/tests/other/base-ui/src/test/java/org/keycloak/testsuite/ui/account2/MyResourcesTest.java:161]: there should always be a link: https://github.com/keycloak/keycloak/blob/0b75961986ccb625fb55a9777fac29dc0315d729/themes/src/main/resources/theme/keycloak-preview/account/resources/app/content/my-resources-page/ResourcesTable.tsx#L130

File: testsuite/integration-arquillian/tests/other/base-ui/src/test/java/org/keycloak/testsuite/ui/account2/page/MyResourcesPage.java (hunk 5 above)
Comment (edewit) [reply testsuite/integration-arquillian/tests/other/base-ui/src/test/java/org/keycloak/testsuite/ui/account2/page/MyResourcesPage.java:148]: oops

File: testsuite/integration-arquillian/tests/other/base-ui/src/test/java/org/keycloak/testsuite/ui/account2/MyResourcesTest.java (hunk 3 above)
Comment (vmuzikar) [reply testsuite/integration-arquillian/tests/other/base-ui/src/test/java/org/keycloak/testsuite/ui/account2/MyResourcesTest.java:161]: What if the client doesn't have Base URL? ;)

**Review by vmuzikar — CHANGES_REQUESTED** [review]
@edewit Thank you for the update.

I believe the last missing part is:
> Please add a test for the icon to check it shows correct number of pending requests.

**Comment by edewit** [comment]
> @edewit Thank you for the update.
> 
> I believe the last missing part is:
> 
> > Please add a test for the icon to check it shows correct number of pending requests.

That is here: https://github.com/keycloak/keycloak/pull/7067/files#diff-7a81d40be58b1db4890e291a1422e5f9R95-R98

**Comment by vmuzikar** [comment]
@edewit Yes but could we perhaps test that the counter actually works, that it doesn't constantly display "1"? :)

**Comment by edewit** [comment]
Well there are only 2 users in the test data, so the max can only be one. Do we have a better way to create more complex test data?

**Comment by vmuzikar** [comment]
@edewit if possible, we should test that it can display something else than "1". So I guess that means we would have to have 3 users I presume. (If it's that much complicated, well, I guess we could settle with testing only "1".)

We should also test that the button is not present at all when there's 0 pending requests.

**Comment by edewit** [comment]
What do you mean? If there are no pending requests (e.g. pending requests are 0) then there is no button displayed

**Comment by vmuzikar** [comment]
Exactly. What I meant was to test that there's no button when there are no pending requests.

**Review by vmuzikar — COMMENTED** [review]
@edewit Thank you for the updates and patience with my requests. :) I think we now have a solid coverage, good work!

LGTM, but one of the tests failed for Chrome:
https://keycloak-jenkins.com/job/keycloak-account2-matrix/897/browser=chrome,jdk=win_oraclejdk8,label=win2016/console
Can you please check it?

**Comment by edewit** [comment]
@vmuzikar they succeeded: https://keycloak-jenkins.com/job/keycloak-account2-matrix/899/

**Review by vmuzikar — DISMISSED** [review]
Ok, it seems they passed in several recent runs (or at least, the failures were unrelated to this). Thanks.

Approving but now there are some conflicts which need to be resolved.

**Comment by edewit** [comment]
yes, because in parallel I started on the polishing of these screens. I'll
rebase and merge these changes stay tuned.

**Comment by vmuzikar** [comment]
@edewit Have you run the tests?

**Comment by edewit** [comment]
UI build: https://keycloak-jenkins.com/job/keycloak-account2-matrix/904/

This also fixes: https://issues.redhat.com/browse/KEYCLOAK-14414

**Comment by edewit** [comment]
UI build: https://keycloak-jenkins.com/job/keycloak-account2-matrix/907/

**Comment by edewit** [comment]
UI build: https://keycloak-jenkins.com/job/keycloak-account2-matrix/908/

**Review by vmuzikar — DISMISSED** [review]
Thanks for the update!

The only thing I'm missing is some simple assert for requestor (and I forgot about scopes too) in the pending requests dialog as we discussed on Zulip, but I guess we can live without it. ;)

Approving...

**Comment by vmuzikar** [comment]
Maybe I'd ask @ssilvert to review the changes in React part.

**Review by ssilvert — CHANGES_REQUESTED** [review]

**Comment by edewit** [comment]
@vmuzikar right added those as well

final final build: https://keycloak-jenkins.com/job/keycloak-account2-matrix/909/

**Review by vmuzikar — DISMISSED** [review]
@edewit Thanks!

From the test perspective LGTM (the test failure is unrelated to this) but as I mentioned – we've got some changes in React part, so I'd be glad if someone with better React knowledge could double check it. :)

**Comment by ssilvert** [comment]
@edewit Did you mean for the storybook artifact to be part of this PR?  If so, we should probably discuss it.

**Comment by edewit** [comment]
@ssilvert that is something I use to develop locally there it was added by mistake.

**Comment by ssilvert** [comment]
> @ssilvert that is something I use to develop locally there it was added by mistake.

Please remove so we can merge.  It's looking like this PR is the key to getting others merged.

**Comment by edewit** [comment]
did that when I commented

**Comment by edewit** [comment]
UI build: https://keycloak-jenkins.com/job/keycloak-account2-matrix/914

**Review by ssilvert — APPROVED** [review]

**Pull request merged**